sunpy/coordinates/utils.py

```diff
@@ -265,3 +265,74 @@ def coordinates(self, points=None):
                         obstime=self.obstime,
                         observer=self.observer,
                         frame=Heliocentric).transform_to(self.start_frame)
+
+
+def get_rectangle_coordinates(bottom_left, *, top_right = None, width: u.deg = None, height: u.deg = None):
+    """
+    Specify a rectangular region of interest in spherical world coordinates.
+
+    Parameters
+    ----------
+    bottom_left : `~astropy.coordinates.BaseCoordinateFrame` or `~astropy.coordinates.SkyCoord`
+        The bottom_left coordinate of the rectangle. If a `SkyCoord` it can
+        have shape ``(2,)`` and also define ``top_right``.
+    top_right : `~astropy.coordinates.BaseCoordinateFrame` or `~astropy.coordinates.SkyCoord`
+        The top_right coordinate of the rectangle. Can only be omitted if
+        ``bottom_left`` has shape ``(2,)``.
+        If in a different frame than ``bottom_left`` and all required metadata for frame conversion is present,
+        ``top_right`` will be transformed to ``bottom_left`` frame.
+        Otherwise a ``ValueError`` will be incurred.
+    width : `~astropy.units.Quantity`
+        The width of the rectangle.
+        Must be omitted if ``bottom_left`` has shape ``(2,)`` or ``top_right`` is provided.
+    height : `~astropy.units.Quantity`
+        The height of the rectangle.
+        Must be omitted if ``bottom_left`` has shape ``(2,)`` or ``top_right`` is provided.
+
+    Returns
+    -------
+    `~astropy.coordinates.BaseCoordinateFrame` or `~astropy.coordinates.SkyCoord`
+        The bottom left coordinate of the rectangular region of interest in spherical world coordinates.
+    `~astropy.coordinates.BaseCoordinateFrame` or `~astropy.coordinates.SkyCoord`
+        The top right coordinate of the rectangular region of interest in spherical world coordinates.
+    """
+    if not (hasattr(bottom_left, 'transform_to') and
+            hasattr(bottom_left, 'shape') and
+            hasattr(bottom_left, 'spherical')):
+        raise TypeError("Invalid input, bottom_left must be of type SkyCoord or BaseCoordinateFrame")
+
+    if (top_right is not None and (hasattr(top_right, 'transform_to') and
+                                   hasattr(top_right, 'shape') and
+                                   hasattr(top_right, 'spherical'))):
+        raise TypeError("Invalid input, top_right if passed, must be of type SkyCoord or BaseCoordinateFrame")
+
+    if bottom_left.shape == (2,) and any((width, height, top_right)):
```

Comment:
I think width and height can be array-like, (a Quantity), which means doing a truthy operation on it would potentially raise an error.

I think it would be safer to do:
```suggestion
    if bottom_left.shape == (2,) and any((x is not None for x in (width, height, top_right))):
```